webrender/src/render_backend.rs

```diff
@@ -120,7 +120,7 @@ impl Document {
         // this code is why we have `Option`, which is never `None`
         let frame_builder = self.frame_ctx.create(
             self.frame_builder.take().unwrap(),
-            &self.scene,
+            &mut self.scene,
```

Comment:
this is worrying

Reply:
it's unfortunately needed to flush the removed pipelines out of the scene. Storing them in the scene itself is part of what helps making sure we don't mess up ordering if we modify a scene and render frames from previous versions. I don't think there is much we can do about the mutability here other than things like Cell which I find more worrying.

Reply:
Perhaps we could keep the list of removed pipelines in the `Document` itself as opposed to the `Scene`? There are two spots where `remove_pipelines` is called, so we aren't going to be doing much extra work, but the semantics of `Scene` would be better then.